packages/bot-web-ui/src/pages/dashboard/cards.tsx

```diff
@@ -25,30 +24,41 @@ type TCardArray = {
 
 const Cards = observer(({ is_mobile, has_dashboard_strategies }: TCardProps) => {
     const { dashboard, load_modal, quick_strategy } = useDBotStore();
-    const {
-        onCloseDialog,
-        dialog_options,
-        is_dialog_open,
-        setActiveTab,
-        setFileLoaded,
-        setPreviewOnPopup,
-        setOpenSettings,
-        showVideoDialog,
-    } = dashboard;
-    const { handleFileChange, loadFileFromLocal } = load_modal;
+    const { onCloseDialog, dialog_options, is_dialog_open, setActiveTab, setPreviewOnPopup } = dashboard;
+    const { toggleLoadModal, setActiveTabIndex } = load_modal;
     const { setFormVisibility } = quick_strategy;
 
-    const [is_file_supported, setIsFileSupported] = React.useState<boolean>(true);
-    const file_input_ref = React.useRef<HTMLInputElement | null>(null);
+    const switchTabOnLoadModalOpen = () => {
+        // Added the setInterval to maintain sequence as tab change was contradicting with load modal open
+        const timerId = setInterval(() => {
```

Comment:
an interval with 0 ms timer? Is there any better way to do this other than this?

Reply:
There is a race condition going on if the strategy is pretty long (more than 2000 lines). Still trying to find a suitable solution but for the time being, using the interval/timeout to keep it out of the main thread so the modal opens first, and then the tab change happens later. Not sure if using `queueMicrotask` would be ideal here. But let me check if there is any other way.

Reply:
can you create a promise that is created when u start going through the strategy and put an await that promise in the `switchTabOnLoadModalOpen()` so that you can wait for it asynchronously instead of trying to put it to the queue indirectly via setIntveral or setTimeout

Reply:
Sure, let me check.